org.adempiere.base/src/org/compiere/model/POInfo.java

```diff
@@ -415,36 +415,31 @@ public boolean isVirtualDBColumn (int index)
 			return true;
 		return m_columns[index].ColumnSQL != null 
 			&& m_columns[index].ColumnSQL.length() > 0
-			&& !m_columns[index].ColumnSQL.startsWith(MColumn.VIRTUAL_UI_COLUMN_PREFIX)
```

Comment:
This seems dangerous, if somebody defines a virtual column as NULL (something strange but I do it sometimes) - then this method will fail.

I suggest the following patch:
[PR2982_pr0.patch.txt](https://github.com/user-attachments/files/23682043/PR2982_pr0.patch.txt)

This works fine and avoids deprecating the methods.

The performance impact doesn't seem too big as MColumn is cached.